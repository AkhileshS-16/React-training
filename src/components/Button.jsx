/* eslint-disable no-unused-vars */
/* eslint-disable react/prop-types */
import React from "react";

const Button = ({ text, className, handleSubmit }) => {
  return (
    <button
      type="submit"
      data-testid="button-test-id"
      className={className}
      onClick={handleSubmit}
    >
      {text}
    </button>
  );
};
export default Button;
