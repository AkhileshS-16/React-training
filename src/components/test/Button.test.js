import React from "react";
import { fireEvent, render } from "@testing-library/react";
import Button from "../Button";

describe("Check if button works", () => {
  test("First", () => {
    const { getByTestId } = render(<Button text="click" />);
    const element = getByTestId("button-test-id");
    expect(element).toBeTruthy();
  });

  test("Second", () => {
    const text = "click";
    const { getByText } = render(<Button text={text} />);
    getByText(text);
  });

  test("check if onClick is triggered", () => {
    const onClick = jest.fn();
    const { getByTestId } = render(<Button handleSubmit={onClick} />);
    const element = getByTestId("button-test-id");
    fireEvent.click(element);
    expect(onClick).toHaveBeenCalled();
  });

  test("Check if snapshots are matched properly", () => {
    const onClick = jest.fn();
    const text = "Click";
    const { asFragment } = render(
      <Button text={text} handleSubmit={onClick} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
