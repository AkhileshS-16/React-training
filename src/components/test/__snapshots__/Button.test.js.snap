// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if button works Check if snapshots are matched properly 1`] = `
<DocumentFragment>
  <button
    data-testid="button-test-id"
    type="submit"
  >
    Click
  </button>
</DocumentFragment>
`;
